plugins/slick.state.js

```diff
@@ -70,13 +70,20 @@
       save();
     }
 
-    function save() {
+    function save(userData) {
       if (_cid && _store) {
         var state = {
           sortcols: getSortColumns(),
           viewport: _grid.getViewport(),
-          columns: getColumns()
+          columns: getColumns(),
+          userData: null
         };
+
+        if (typeof userData !== typeof undefined)
```

Comment:
I've never seen it written this way (I mean `typeof undefined`), but why not just write it like `if (userData !== undefined) {` it works in all browsers nowadays. Your way of writing it also works, it's just a bit longer

Reply:
true, it's a coding habit I have gotten used to for compatibility reasons (some of our customers use really old browsers...)

Reply:
just remember that  ```userData !== undefined``` with throw an error if ```userData``` us undeclared, where ```typeof userData !== 'undefined'``` won't